doc/explanation/api/reactivity.md

```diff
@@ -0,0 +1,177 @@
+
+```python
+pn.param.ParamFunction(pn.bind(add, a_slider.param.value, b_slider.param.value), inplace=True)
+```
+:::
+
+Function level binding provides significantly flexibility, e.g. we can write reactive components that change type depending on the input. We could return a different component depending on the value of some widgets and Panel would re-render the output appropriately. However this also means that compared to component level binding, function level binding is significantly less efficient and where possible component level binding should be preferred.
+
+However, if we think back to the diagram above, we can hopefully immediately see the benefit of this reactive approach, whether component or function. In a callback based approach it is immediately unclear how we would combine the state of `a_slider` and `b_slider` into a single update. Either we have to resort to accessing values directly introducing unseen entanglement of our UI code and logic, quickly resulting in spaghetti code or we have introduce a class that acts as a controller to track the updates and hold the state.
+
+## Reactive References
+
+So far we have explored the difference between reactive and callback based code and gone over component-level and function-level binding to achieve reactivity in Panel. To really understand the power of binding we have to dive a little deeper into what it references are and how they can be bound. In particular we have, so far, focused primarily on Parameters as references. This makes sense since pretty much all reactivity in Panel is ultimately driven by a parameter. However binding parameters on their own is usually not that useful. Thinking back to our simple component-level binding example:
+
+```python
+text = pn.widgets.TextInput(value='Hello world!')
+
+pn.pane.Markdown(object=text.param.value)
+```
+
+You might have wondered: "why is this useful?". You rarely want to bind the output of a widget directly to some other output or parameter. In practice an output or parameter value will usually be the result of transforming one or more inputs. For this reason Panel (and Param, which it builds on) allows a number of different types to be used as a reference:
+
+1. Parameters: A `Parameter` object is the simplest reference.
+2. Widgets: A Panel `Widget` acts as a proxy for its `value` parameter, i.e. passing `text` and `text.param.value` are equivalent in the last example.
+3. Bound functions: A function which has other references bound to it using the `pn.bind` helper may also be used as a reference.
+4. Reactive Expressions: Param offers so called reactive expressions which act as proxies for their underlying objects and can be chained into arbitrary expressions.
+5. Async Generators: Asynchronous generators functions can be used as a reference to drive streaming outputs.
```

Comment:
Would it be possible to add 5 tabs with e2e runnable code. Each tab illustrating one of the 5 different reference types?